hibernate/cache-commons/src/main/java/org/infinispan/hibernate/cache/commons/util/Caches.java

```diff
@@ -225,8 +229,10 @@ public static void broadcastEvictAll(AdvancedCache cache) {
 		final RpcManager rpcManager = cache.getRpcManager();
 		if ( rpcManager != null ) {
 			final EvictAllCommand cmd = new EvictAllCommand(ByteString.fromString(cache.getName()));
-			final RpcOptions options = rpcManager.getDefaultRpcOptions(isSynchronousCache(cache));
-			rpcManager.invokeRemotely(null, cmd, options);
+			final RpcOptions options = isSynchronousCache(cache) ? rpcManager.getSyncRpcOptions() :
+					new RpcOptions(DeliverOrder.PER_SENDER, cache.getCacheConfiguration().clustering().remoteTimeout(), TimeUnit.MILLISECONDS);
```

Comment:
suggestion: create `RpcManager.getSyncOrderedRpcOptions();` if this method is invoked too often.

Reply:
I think we should do that if the number of uses increased. At the moment, it's only used twice throughout the codebase AFAICT.